### PR TITLE
Add compileBuiltIn property to project.json

### PR DESCRIPTION
**Pull request by troydai**
#1288

### CODE REVIEW

**Comment by troydai** [comment]
@glennc @victorhurdugaci @lodejard @ChengTian 
Please help to take a look.

File: src/Microsoft.Framework.Runtime.Hosting/PatternGroup.cs [hunk 1]
```diff
@@ -33,7 +33,15 @@ internal PatternGroup(IEnumerable<string> includePatterns, IEnumerable<string> e
             _matcher.AddExcludePatterns(ExcludePatterns);
         }
 
-        public static PatternGroup Build(JObject rawProject, string projectDirectory, string projectFilePath, string name, string legacyName, IEnumerable<string> fallback, IEnumerable<string> buildInExcludePatterns, bool includePatternsOnly = false)
+        public static PatternGroup Build(JObject rawProject,
+                                         string projectDirectory,
+                                         string projectFilePath,
+                                         string name,
+                                         string legacyName,
+                                         IEnumerable<string> fallbackIncludeing = null,
```
Comment (Praburaj) [inline src/Microsoft.Framework.Runtime.Hosting/PatternGroup.cs]: typo in `fallbackIncludeing`

Comment (troydai) [inline src/Microsoft.Framework.Runtime.Hosting/PatternGroup.cs]: Will fix. Thanks.

**Comment by troydai** [comment]
@davidfowl

**Comment by davidfowl** [comment]
This shouldn't break people that respecify the pattern right?

**Comment by troydai** [comment]
No, at least for all projects in universe. But there is possibility that it will breaks, if user explicitly overwrite `"code"` to not include all `**\*.cs`. We can't avoid breaking them.

**Comment by troydai** [comment]
@davidfowl sign off?

Ping

File: test/Microsoft.Framework.Runtime.Tests/FileGlobbing/ProjectFilesCollectionFacts.cs [hunk 2]
```diff
@@ -88,6 +89,20 @@ public void FilesPatternsAreSet()
             Assert.Equal(new string[] { "buggy/*.*", "bin/**", "obj/**", "**/*.kproj", "no_pack/*.*" }, target.ContentPatternsGroup.ExcludePatterns);
         }
 
+        [Fact]
+        public void RewriteCompileBuiltIn()
```
Comment (davidfowl) [inline test/Microsoft.Framework.Runtime.Tests/FileGlobbing/ProjectFilesCollectionFacts.cs]: Add a test that does:

```JSON
{
  "compile": ["**/*.cs"]
}
```

**Comment by troydai** [comment]
Two tests added.

**Comment by troydai** [comment]
Ping

**Pull request merged**